server/server.go

```diff
@@ -470,7 +461,7 @@ func (j *jsonRPCHub) GetForksInTime(blockNumber uint64) chain.ForksInTime {
 }
 
 func (j *jsonRPCHub) GetStorage(stateRoot types.Hash, addr types.Address, slot types.Hash) ([]byte, error) {
-	account, err := j.getAccountImpl(stateRoot, addr)
+	account, err := getAccountImpl(j.state, stateRoot, addr)
```

Comment:
What do you think about returning `snapshot` instance from the `getAccountImpl`? Asking this, because in this use case, we are going to call `NewSnapshotAt` twice for the same root. 

Although it probably doesn't matter, as we rely on LRU cache when accessing to the trie, and when we call `NewSnapshotAt` second time, we will most probably have the trie in the cache.

Reply:
The fact that it is cached from the previous call in getAccountImpl, and that only the GetStorage (which is not a frequently used endpoint) I think it's better to make a compromise to have a smaller method signature in this case.

Reply:
Yep agreed, just was thinking out loud and wasn't sure myself either :+1: